xayn-ai/src/embedding/utils.rs

```diff
@@ -67,11 +69,10 @@ where
 /// [`TrustedLen`]: std::iter::TrustedLen
 /// [`ExactSizeIterator`]: std::iter::ExactSizeIterator
 /// [`Chain`]: std::iter::Chain
-pub fn pairwise_cosine_similarity<I, S>(iter: I) -> Array2<f32>
+pub fn pairwise_cosine_similarity<'a, I>(iter: I) -> Array2<f32>
 where
-    I: IntoIterator<Item = ArrayBase<S, Ix1>>,
+    I: IntoIterator<Item = ArrayView1<'a, f32>>,
```

Comment:
I made this change to simplify the signatures. I deemed this reasonable because, on any owned representation, calling `.view()` will convert it to `ArrawView`.